onnxruntime/core/platform/posix/env.cc

```diff
@@ -220,14 +224,13 @@ class PosixThread : public EnvThread {
         cpu_set_t cpuset;
         CPU_ZERO(&cpuset);
         CPU_SET(p->affinity_mask, &cpuset);
-        // We have no way of reporting or logging affinity failure
-        // and we are not at liberty of terminating the thread
-        // because it would make things worse.
-        pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
-        //if (s != 0) {
-        //  auto [err_no, err_msg] = GetSystemError();
-        //  ORT_THROW("pthread_setaffinity_np failed, error code: ", err_no, " error msg: ", err_msg);
-        //}
+        // pthread_setaffinity_np() does not set errno, it returns it.
+        auto ret = pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
+        if (ret != 0) {
+          auto [err_no, err_msg] = GetSystemError(ret);
+          LOGS_DEFAULT(ERROR) << "pthread_setaffinity_np failed for thread: " << gettid()
```

Comment:
WARNING since we're continuing any way and the user can't really do anything on their end to fix this?

Reply:
I am going to suggest to explicitly specify the number of threads or correct affinity masK?